x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/evaluate_panel.tsx

```diff
@@ -0,0 +1,299 @@
+  >([]);
+  const [columns, setColumns] = useState<any>([]);
+  const [columnsData, setColumnsData] = useState<any>([]);
+  const [popoverContents, setPopoverContents] = useState<any>([]);
+  const [docsCount, setDocsCount] = useState<null | number>(null);
+  const [error, setError] = useState<null | string>(null);
+
+  // Column visibility
+  const [visibleColumns, setVisibleColumns] = useState(() =>
+    columns.map(({ id }: { id: string }) => id)
+  );
+
+  const index = jobConfig.dest.index;
+  const dependentVariable = getDependentVar(jobConfig.analysis);
+  const predictionFieldName = getPredictionFieldName(jobConfig.analysis);
+  // default is 'ml'
+  const resultsField = jobConfig.dest.results_field;
+
+  const loadData = async ({
+    isTrainingClause,
+    ignoreDefaultQuery = true,
+  }: {
+    isTrainingClause: { query: string; operator: string };
+    ignoreDefaultQuery?: boolean;
+  }) => {
+    setIsLoading(true);
+
+    const evalData = await loadEvalData({
+      isTraining: false,
+      index,
+      dependentVariable,
+      resultsField,
+      predictionFieldName,
+      searchQuery,
+      ignoreDefaultQuery,
+      jobType: ANALYSIS_CONFIG_TYPE.CLASSIFICATION,
+    });
+
+    const docsCountResp = await loadDocsCount({
+      isTraining: false,
+      searchQuery,
+      resultsField,
+      destIndex: jobConfig.dest.index,
+    });
+
+    if (
+      evalData.success === true &&
+      evalData.eval &&
+      isClassificationEvaluateResponse(evalData.eval)
+    ) {
+      const confusionMatrix =
+        evalData.eval?.classification?.multiclass_confusion_matrix?.confusion_matrix;
+      setError(null);
+      setConfusionMatrixData(confusionMatrix || []);
+      setIsLoading(false);
+    } else {
+      setIsLoading(false);
+      setConfusionMatrixData([]);
+      setError(evalData.error);
+    }
+
+    if (docsCountResp.success === true) {
+      setDocsCount(docsCountResp.docsCount);
+    } else {
+      setDocsCount(null);
+    }
+  };
+
+  useEffect(() => {
+    if (confusionMatrixData.length > 0) {
+      const { columns: derivedColumns, columnData } = getColumnData(confusionMatrixData);
+      // Initialize all columns as visible
+      setVisibleColumns(() => derivedColumns.map(({ id }: { id: string }) => id));
+      setColumns(derivedColumns);
+      setColumnsData(columnData);
+      setPopoverContents({
+        numeric: ({
+          cellContentsElement,
+          children,
+        }: {
+          cellContentsElement: any;
+          children: any;
+        }) => {
+          const rowIndex = children?.props?.rowIndex;
+          const colId = children?.props?.columnId;
+          const gridItem = columnData[rowIndex];
+          const count = colId === gridItem.actual_class ? gridItem.count : gridItem.error_count;
+
+          return `${count} / ${gridItem.actual_class_doc_count} * 100 = ${cellContentsElement.textContent}`;
+        },
+      });
+    }
+  }, [confusionMatrixData]);
```

Comment:
To do a deep equal, maybe `JSON.stringify(confusionMatrixData)` is worth doing here similar to `JSON.stringify(searchQuery)` further down.